src/main/java/seedu/address/commons/core/index/IndexList.java

```diff
@@ -0,0 +1,93 @@
+package seedu.address.commons.core.index;
+
+import java.util.ArrayList;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Represents an array of zero-based or one-based indices.
+ * The indices in the array are distinct.
+ */
+public class IndexList {
+    private final ArrayList<Index> indices;
+
+    /**
+     * IndexList can only be created by calling {@link IndexList#emptyList()}.
+     */
+    private IndexList() {
+        indices = new ArrayList<>();
+    }
+
+    /**
+     * Creates a new {@code IndexList} with an empty list.
+     */
+    public static IndexList emptyList() {
+        return new IndexList();
+    }
+
+    /**
+     * Adds a new Index into the list if the {@code Index} is unique.
+     *
+     * @param index a new index to be added into the list.
+     */
+    public void add(Index index) {
+        if (!indices.contains(index)) {
+            indices.add(index);
+        }
+    }
+
+    /**
+     * Returns the current size of the index list.
+     *
+     * @return integer representing the number of elements in the list.
+     */
+    public int size() {
```

Comment:
Method name must be verb. (But to be fair the underlying class name is also `size`)